lib/game.rb

```diff
@@ -0,0 +1,249 @@
+class Game
+  attr_accessor :player_1, :player_2, :board
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+  ]
+
+  TRAPS = [
+    [0,4,2],
+    [0,4,8],
+    [2,4,8],
+    [4,6,8],
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    # accepts 2 players and a board
+    # defaults to two human players, X and O, with an empty board
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def cells
+    @board.cells
+  end
+
+  def turn_count
+  count = 0
+  cells.each do |play|
+    if play == "O" || play == "X"
+      count +=1
+    end
+  end
+  count
+end
+
+  def current_player
+    # returns the correct player, X, for the third move
+    if turn_count.even?
+      player_1
+    else
+      player_2
+    end
+  end
+
+  def won?
+    # returns false for a draw
+    # returns the correct winning combination in the case of a win
+    # isn't hard-coded
+    winner = nil
+    WIN_COMBINATIONS.each do |combo|
+      if combo.all? {|idx| cells[idx] == "X"}
+        winner = combo
+      elsif combo.all? {|idx| cells[idx] == "O"}
+        winner = combo
+      else
+        false
+      end
+    end
+    if winner != nil
+      winner
+    else
+      false
+    end
+  end
+
+  def full?
+    cells.all? {|token| token == "X" || token == "O"}
+  end
+
+  def draw?
+    # returns true for a draw
+    # returns false for a won game
+    # returns false for an in-progress game
+    if full? && !(won?)
+      true
+    else
+      false
+    end
+  end
+
+  def over?
+    # returns true for a draw
+    # returns true for a won game
+    # returns false for an in-progress game
+    if draw? || won? || full?
+      true
+    else
+      false
+    end
+  end
+  def winner
+    # returns X when X won
+    # returns O when O won
+    # returns nil when no winner
+    WIN_COMBINATIONS.each do |combo|
+      if combo.all? {|idx| cells[idx] == "X"}
+        return "X"
+      elsif combo.all? {|idx| cells[idx] == "O"}
+        return "O"
+      end
+    end
+    nil
+  end
```

Comment:
Use the `#won?` method you've already built.
```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.